common-util/ShowBadge.tsx

```diff
@@ -7,13 +7,19 @@ import styled from 'styled-components';
 
 import { isGoerli } from '@autonolas/frontend-library';
 
+import { SetupState } from 'types/hooks';
+
 import { getAutonolasTokenUri } from '../components/Leaderboard/MintNft/utils';
 
 const IMAGE_SIZE = 160;
 const { Text } = Typography;
 
-export const ShowBadge = ({ image, tokenId }) => {
-  const chainId = useSelector((state) => state?.setup?.chainId);
+export const ShowBadge = ({ image, tokenId }: { image: string; tokenId: string }) => {
+  const chainId = useSelector((state: SetupState) => state?.setup?.chainId);
```

Comment:
ideally we should use useAppSelector that is defined inside `store/index.ts` and exported from there, so that the types are automatically known. here's an example from Govern https://github.com/valory-xyz/autonolas-frontend-mono/blob/main/apps/govern/store/index.ts#L19
let me know if there are difficulties with using this approach, we can jump on a call and discuss